lua/mini/misc.lua

```diff
@@ -409,6 +409,139 @@ MiniMisc.zoom = function(buf_id, config)
+
+  if not H.is_array_of(opts.ignore_buftype, H.is_string) then
+    H.error('In `setup_restore_cursor()` `opts.ignore_buftype` should be an array of strings.')
+  end
+
+  if not H.is_array_of(opts.ignore_filetype, H.is_string) then
+    H.error('In `setup_restore_cursor()` `opts.ignore_filetype` should be an array of strings.')
+  end
+
+  if not H.is_boolean(opts.center) then
+    H.error('In `setup_restore_cursor()` `opts.center` should be a boolean.')
+  end
+
+  -- local augroup = vim.api.nvim_create_augroup('MiniMiscRestoreCursor',
+  --                                             { clear = true })
+  -- local au_command = string.format("lua require('mini.misc').restore_cursor_buftype(%s)",
+  --                                  vim.inspect(opts, { newline = ' ', indent = '' }))
+  -- vim.api.nvim_create_autocmd(
+  --   'BufReadPre',
+  --   {
+  --     group = augroup,
+  --     command = au_command,
+  --     desc = 'Setup the restore cursor autocmd',
+  --   })
+
+  local au_command = string.format(
+    [[augroup MiniMiscRestoreCursor
+      au!
+      au BufReadPre * au FileType <buffer> ++once lua require('mini.misc').restore_cursor(%s)
+    augroup END]],
+    vim.inspect(opts, { newline = ' ', indent = '' }))
+  vim.api.nvim_exec(au_command, false)
+end
+
+-- MiniMisc.restore_cursor_buftype = function(opts)
+--   -- Check if the buffer should be ignored
+--   if vim.tbl_contains(opts.ignore_buftype,
+--                       vim.api.nvim_buf_get_option(0, "buftype")) then
+--     return
+--   end
+--
+--   local augroup = vim.api.nvim_create_augroup('MiniMiscRestoreCursor',
+--                                               { clear = false })
+--   local au_command = string.format("lua require('mini.misc').restore_cursor(%s)",
+--                                    vim.inspect(opts, { newline = ' ', indent = '' }))
+--   vim.api.nvim_create_autocmd(
+--     'FileType',
+--     {
+--       group = augroup,
+--       command = au_command,
+--       buffer = vim.api.nvim_get_current_buf(),
+--       once = true,
+--       desc = 'Restore the cursor position',
+--     })
+-- end
+
+--- Restore cursor when reopening a file
+MiniMisc.restore_cursor = function(opts)
+  -- Check if the buffer should be ignored
+  if vim.tbl_contains(opts.ignore_buftype,
+                      vim.api.nvim_buf_get_option(0, "buftype")) then
+    return
+  end
+
+  -- Check if the filetype should be ignored
+  if vim.tbl_contains(opts.ignore_filetype,
+                      vim.api.nvim_buf_get_option(0, "filetype")) then
+      -- reset cursor to first line
+      vim.api.nvim_command([[normal! gg]])
+      return
+  end
+
+  -- Place cursor --
+
+  -- If a line has already been specified on the command line, we are done
+  --   nvim file +num
+  if vim.fn.line(".") > 1 then
+      return
+  end
+
+  local last_line = vim.fn.line([['"]])
+  local buff_last_line = vim.fn.line("$")
+
+  if last_line > 0 and last_line <= buff_last_line then
```

Comment:
This can be a separate early return block.

Is there a reason to use `last_line > 0` instead of `last_line > 1` as in docs? The latter seems to ensure that cursor is not restored to the same place it is at right now.

Reply:
You're right, and early return is nicer.

Fixed.